pkg/asset/tls/utils.go

```diff
@@ -54,3 +54,17 @@ func PublicKeyToPem(key *rsa.PublicKey) (string, error) {
 	)
 	return string(keyinPem), nil
 }
+
+// PemToPrivateKey converts a data block to rsa.PrivateKey.
+func PemToPrivateKey(data []byte) (*rsa.PrivateKey, error) {
+	block, _ := pem.Decode(data)
+	key, err := x509.ParsePKCS1PrivateKey(block.Bytes)
+	return key, err
```

Comment:
nit: these last two lines could be collapsed to:

```go
return x509.ParsePKCS1PrivateKey(block.Bytes)
```

(and similarly in `PemToCertificate` below).